mosec/coordinator.py

```diff
@@ -124,14 +135,21 @@ def run(self):
             self.coordinate()
 
     def get_decoder(self) -> Callable:
+        deserialize_ipc = self.worker._deserialize_ipc
         if STAGE_INGRESS in self.worker._stage:
-            return self.worker.deserialize
-        return self.worker._deserialize_ipc
+            return lambda batch: [self.worker.deserialize(x) for x in batch]
+        if self.shm_wrapper is not None:
+            wrapped_decode = self.shm_wrapper.wrap_decoder(deserialize_ipc)
+            return lambda batch: wrapped_decode(batch)
+        return lambda batch: [deserialize_ipc(x) for x in batch]
```

Comment:
The lambda function here looks super weird.

I wonder if the IPC wrapper can be separated from the de/serialization layer, so we don't need to change the encode/decode behavior (batch or single). Since we can get the ingress/egress information.

Reply:
No the main reason for batch is because for plasma, batch get is potentially more efficient. So have to pass a list to plasma client get, thus must put a list compression somewhere. The lambda here tries to do that.

Reply:
It's necessary when you put the wrapper into the encoder/decoder. But it should be another layer that handles the payloads.

Reply:
Any suggestion?